src/parser/semantic_check.c

```diff
@@ -8497,17 +8497,36 @@ pt_check_create_entity (PARSER_CONTEXT * parser, PT_NODE * node)
   qry_specs = node->info.create_entity.as_query_list;
   if (node->info.create_entity.entity_type == PT_CLASS)
     {
-      PT_NODE *select = NULL;
+      PT_NODE *select = node->info.create_entity.create_select;
+      PT_NODE *crt_attr = NULL;
+      PT_NODE *const qspec_attr = pt_get_select_list (parser, select);
+
       /* simple CLASSes must not have any query specs */
       if (qry_specs)
 	{
 	  PT_ERRORm (parser, qry_specs, MSGCAT_SET_PARSER_SEMANTIC, MSGCAT_SEMANTIC_CLASS_NO_QUERY_SPEC);
 	  return;
 	}
       /* user variables are not allowed in select list for CREATE ... AS SELECT ... */
-      select = node->info.create_entity.create_select;
       if (select != NULL)
 	{
+	  for (crt_attr = qspec_attr; crt_attr != NULL; crt_attr = crt_attr->next)
+	    {
+	      PT_NODE *s_attr = NULL;
+
+	      if (crt_attr->node_type == PT_NAME || crt_attr->alias_print)
+		{
+		  s_attr = crt_attr;
+		}
+
+	      if (s_attr == NULL)
```

Comment:
how about?
```	      
if (crt_attr->node_type != PT_NAME && crt_attr->alias_print  == NULL)
```

Reply:
Ok, it seems that's better readability.

Reply:
How about checking PT_EXPR? I think PT_VALUE is okay.
```
if (crt_attr->node_type == PT_EXPR && crt_attr->alias_print  == NULL)
```

Reply:
PT_DOT_ is also OK.

Reply:
I think it should be excluded PT_NAME and PT_DOT_ only.

Reply:
OK. I agree only PT_NAME.